src/rules.jl

```diff
@@ -5,46 +5,46 @@
 # unary #
 #-------#
 
-@define_diffrule Base.:+(x)                   = :(  1                                  )
-@define_diffrule Base.:-(x)                   = :( -1                                  )
+@define_diffrule Base.:+(x)                   = :(  one($x)                            )
+@define_diffrule Base.:-(x)                   = :( -one($x)                            )
```

Comment:
Is this needed? The derivative does not necessarily have exactly the same typr as `x`, I guess we mainly want to avoid unnecessary promotions. But `1` and `-1` already seem quite safe and shouldn't cause any promotions?

Reply:
Sure, good point :+1: I'll revert this